### PR TITLE
[pwa] clean up event statuses on event appearances, add TeamLink

### PR DESCRIPTION
**Pull request by tervay**
District points needs some UI cleanup but it's a start.

Also adds TeamLink and EventLink components to make lives easier

<details>
<summary>Screenshots</summary>

image

image

image

</details>

### CODE REVIEW

File: pwa/app/routes/team.$teamNumber.($year).tsx [hunk 1]
```diff
@@ -120,6 +124,32 @@ async function loadData(params: Params) {
     throw new Response(null, { status: 404 });
   }
 
+  // TODO: fetch these in parallel after initial render
```
Comment (fangeugene) [inline pwa/app/routes/team.$teamNumber.($year).tsx:127]: what is the proposed mechanism for doing this? ReactQuery?

Comment (tervay) [reply pwa/app/routes/team.$teamNumber.($year).tsx:127]: Yeah, most likely

File: pwa/app/components/tba/teamEventAppearance.tsx [hunk 2]
```diff
@@ -53,11 +76,172 @@ export default function TeamEventAppearance({
...
+                      r.awardee !== '' &&
+                      r.team_key === team.key,
+                  );
+
+                  return (
+                    <li key={`${award.award_type}-${award.event_key}`}>
+                      {award.name}
+                      {namedRecipients.length > 0 ? (
+                        <>
+                          {' '}
```
Comment (fangeugene) [inline pwa/app/components/tba/teamEventAppearance.tsx:145]: Is this leading space necessary?

File: pwa/app/components/tba/teamEventAppearance.tsx [hunk 3]
```diff
@@ -53,11 +76,172 @@ export default function TeamEventAppearance({
...
+              {status.alliance.name}
+            </dt>
+            <dd className="text-gray-700 sm:col-span-2">
+              {joinComponents(
+                maybeAlliances
+                  .find((a) => a.picks.includes(team.key))
+                  ?.picks.map((k) => (
+                    <Link
+                      key={k}
+                      to={`/team/${k.substring(3)}`}
```
Comment (fangeugene) [inline pwa/app/components/tba/teamEventAppearance.tsx:173]: `.substring(3)` might break for "B" teams? On prod we also strip the "B" out and just link to the base team.

File: pwa/app/components/tba/teamEventAppearance.tsx (hunk 2 above)
Comment (tervay) [reply pwa/app/components/tba/teamEventAppearance.tsx:145]: Yes, it creates the space between the award and the names:

image

File: pwa/app/components/tba/teamEventAppearance.tsx (hunk 3 above)
Comment (tervay) [reply pwa/app/components/tba/teamEventAppearance.tsx:173]: Ugh, we really just need a `<TeamLink />` component to handle things

Comment (fangeugene) [reply pwa/app/components/tba/teamEventAppearance.tsx:173]: I don't think this needs to be dealt with now, but we need to do a pass on B team handling at some point.

**Review by fangeugene — APPROVED** [review]

File: pwa/app/components/tba/teamEventAppearance.tsx (hunk 3 above)
Comment (tervay) [reply pwa/app/components/tba/teamEventAppearance.tsx:173]: I have a solve nearly done

File: pwa/app/components/tba/allianceSelectionTable.tsx [hunk 4]
```diff
@@ -80,7 +81,7 @@ export default function AllianceSelectionTable(props: {
         <TableBody>
           {props.alliances.map((a, idx) => (
             <AllianceTableRow
-              key={a.name}
+              key={a.name ?? `alliance-${idx}`}
```
Comment (fangeugene) [inline pwa/app/components/tba/allianceSelectionTable.tsx:84]: Should we just make the key the index? i don't think the test for name and fallback to index is necessary.

**Review by fangeugene — APPROVED** [review]
minor nit otherwise lg

**Review by fangeugene — APPROVED** [review]

**Pull request merged**